Dev

cyberschoolinmoscow opened the pull request:
Задача спринта №8

cyberschoolinmoscow committed:
Изменена тема с Material3 на MaterialComponents

cyberschoolinmoscow committed:
Изменена тема с Material3 на MaterialComponents для values-night

cyberschoolinmoscow committed:
Добавлены layout\activity_main, иконки для кнопок

cyberschoolinmoscow committed:
Добавлены layout\activity_settings и values\styles

cyberschoolinmoscow committed:
Добавлены пустые MediatekaActivity и SearchActivity, переходы из MainActivity в MediatekaActivity, SearchActivity, SettingsActivity, переход из SettingsActivity в MainActivity.

ArturNurtdinov commented on app/src/main/java/com/practicum/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,46 @@
         setContentView(R.layout.activity_main)
+        myToolbar = findViewById(R.id.idMainToolbar)
+        setSupportActionBar(myToolbar)
+        val buttonSearch = findViewById<MaterialButton>(R.id.idSearch)
+        val buttonMediateka = findViewById<MaterialButton>(R.id.idMediateka)
+        val buttonSettings = findViewById<MaterialButton>(R.id.idSettings)
```
Переменные лучше объявлять максимально близко к их первому использованию

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,57 @@
+
+            <ImageView
+                style="@style/settings_image"
+                android:src="@drawable/offswitch" />
```
Здесь нужно будет использовать `SwitchCompat` в качестве вью, т.к. в дальнейшем придётся обрабатывать переключение темы. То есть это не просто иконка - это интерактивный элемент на экране, который проще всего сделать через `SwitchCompat`

ArturNurtdinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,29 @@
 
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+    <com.google.android.material.button.MaterialButton
```
У первой кнопки от текста в шапке отступ в 36dp, сейчас немного меньше на главном экране, рекомендую поправить

ArturNurtdinov commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,10 @@
 <resources>
-    <string name="app_name">Playlist Maker</string>
+    <string name="app_name">Playlist maker</string>
```
Здесь можно указать свойство `translatable="false"`, т.к. эта строчка не требует перевода

ArturNurtdinov commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,59 @@
+        <item name="android:textAllCaps">false</item>
+        <item name="android:layout_marginHorizontal">@dimen/margin_horizontal</item>
+        <item name="android:layout_marginBottom">@dimen/margin_vertical</item>
+        <item name="android:layout_height">match_parent</item>
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_weight">1</item>
```
Если вес элемента = 1 и мы хотим, чтобы кнопка растягивалась, то лучше высоту указать как 0dp - это специальное значение, говорящее о том, что кнопке нужно будет занять максимально доступное место по высоте

cyberschoolinmoscow committed:
1) Созданы общие стили для атрибутов на «Главном экране» и в «Настройках».
2) Добавлены новые шрифты в экраны.
3) Адаптирован интерфейс под «Ночной режим».
4) При включении ночного режима приложение меняет тему на тёмную.
5) Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».
6) Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.
7) Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

cyberschoolinmoscow commented:
1
